### PR TITLE
Utility function for loading Discord chunks

### PR DESCRIPTION
**Pull request by Nuckyz**

### CODE REVIEW

File: src/webpack/webpack.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export const filters = {
 export const subscriptions = new Map<FilterFn, CallbackFn>();
 export const listeners = new Set<CallbackFn>();
 
-export type CallbackFn = (mod: any, id: number) => void;
+export type CallbackFn = (mod: any, id: string) => void;
```
Comment (Vendicated) [inline src/webpack/webpack.ts:72]: why this change?

File: src/webpack/webpack.ts [hunk 2]
```diff
@@ -201,43 +201,53 @@ export const findBulk = traceFunction("findBulk", function findBulk(...filterFns
 
     if (found !== length) {
         const err = new Error(`Got ${length} filters, but only found ${found} modules!`);
-        if (IS_DEV) {
-            if (!devToolsOpen)
-                // Strict behaviour in DevBuilds to fail early and make sure the issue is found
-                throw err;
-        } else {
-            logger.warn(err);
-        }
```
Comment (Vendicated) [inline src/webpack/webpack.ts:210]: why did you change this code? the warning is kinda annoying when you're testing filters in devtools, hence why this is here

File: src/webpack/webpack.ts [hunk 3]
```diff
@@ -201,43 +201,53 @@ export const findBulk = traceFunction("findBulk", function findBulk(...filterFns
...
-export const lazyWebpackSearchHistory = [] as Array<["find" | "findByProps" | "findByCode" | "findStore" | "findComponent" | "findComponentByCode" | "findExportedComponent" | "waitFor" | "waitForComponent" | "waitForStore" | "proxyLazyWebpack" | "LazyComponentWebpack", any[]]>;
+/**
+ * Find the first module factory that includes all the given code
+ * @returns The module factory or null
+ */
+export function findModule(...code: string[]) {
+    const id = findModuleId(...code);
+    if (!id) return null;
+
+    return wreq.m[id].original ?? wreq.m[id];
```
Comment (Vendicated) [inline src/webpack/webpack.ts:247]: why? we do not want people to use the original ones

File: src/webpack/webpack.ts [hunk 4]
```diff
@@ -272,26 +282,28 @@ export function LazyComponentWebpack<T extends object = any>(factory: () => any,
...
-/**
- * Find the first module that has the specified properties
- */
-export function findByProps(...props: string[]) {
-    const res = find(filters.byProps(...props), { isIndirect: true });
+function indirectFind(name: string, filter: FilterFn) {
+    const res = find(filter, { isIndirect: true });
     if (!res)
-        handleModuleNotFound("findByProps", ...props);
+        handleModuleNotFound(name, filter);
```
Comment (Vendicated) [inline src/webpack/webpack.ts:296]: this goes entirely against the point. you're now logging the `filters.byFoo()` return value which is identical for every filter and has 0 meaningful info unless you check the closure

File: src/webpack/webpack.ts (hunk 3 above)
Comment (Nuckyz) [reply src/webpack/webpack.ts:247]: oh that's true, the issue is it's returning our factory method instead of the mod, we isnt what we want

File: src/webpack/webpack.ts (hunk 1 above)
Comment (Nuckyz) [reply src/webpack/webpack.ts:72]: to not cast to number so much, since when we get the key it's always a string

File: src/webpack/webpack.ts (hunk 2 above)
Comment (Nuckyz) [reply src/webpack/webpack.ts:210]: sorry I was confused on why it didnt warn in dev mode

**Review by FieryFlames — APPROVED** [review]
